seaborn/categorical.py

```diff
@@ -1942,41 +1953,47 @@ def horz_perc_box(x, b, i, k, w):
             hex_color = mpl.colors.rgb2hex(color)
 
             if vert:
-                boxes = [vert_perc_box(x, b[0], i, k, b[1])
-                         for i, b in enumerate(zip(box_ends, w_area))]
-
-                # Plot the medians
-                ax.plot([x - widths / 2, x + widths / 2], [y, y],
-                        c='.15', alpha=.45, **kws)
+                box_func = vert_perc_box
+                xs_median = [x - widths / 2, x + widths / 2]
+                ys_median = [y, y]
+                xs_outliers = np.full(len(outliers), x)
+                ys_outliers = outliers
 
-                ax.scatter(np.repeat(x, len(outliers)), outliers,
-                           marker='d', c=hex_color, **kws)
             else:
-                boxes = [horz_perc_box(x, b[0], i, k, b[1])
-                         for i, b in enumerate(zip(box_ends, w_area))]
+                box_func = horz_perc_box
+                xs_median = [y, y]
+                ys_median = [x - widths / 2, x + widths / 2]
+                xs_outliers = outliers
+                ys_outliers = np.full(len(outliers), x)
 
-                # Plot the medians
-                ax.plot([y, y], [x - widths / 2, x + widths / 2],
-                        c='.15', alpha=.45, **kws)
+            boxes = [box_func(x, b[0], i, k, b[1])
+                     for i, b in enumerate(zip(box_ends, w_area))]
 
-                ax.scatter(outliers, np.repeat(x, len(outliers)),
-                           marker='d', c=hex_color, **kws)
+            # Plot the medians
+            with axes_style(rc={"lines.solid_capstyle": "butt"}):
```

Comment:
This parameter can go directly to `plot`
e.g.
```python
f, ax = plt.subplots()
ax.plot([1, 1], lw=10, solid_capstyle="butt")
```
Unless I'm missing something.

Reply:
Will change this. 
My original intent was allowing the user overriding this through kwargs, but since `plot` and `scatter` kwargs are shared this is not possible anyway. And perhaps we don't even want users to change this?

Reply:
Indeed, lack of full-customization via specific kwargs is a problem throughout the categorical module. Could be improved, but doesn't need to happen here.